packages/rest/src/rest.ts

```diff
@@ -158,6 +161,50 @@ export function getDiscriminator(program: Program, entity: Type): Discriminator
   return undefined;
 }
 
+// ----------------- @autoRoute -----------------
+
+const autoRouteKey = createStateSymbol("autoRoute");
+
+/**
+ * `@autoRoute` enables automatic route generation for an operation, namespace, or interface.
+ *
+ * When applied to an operation, it automatically generates the operation's route based on path parameter
+ * metadata.  When applied to a namespace or interface, it causes all operations under that scope to have
+ * auto-generated routes.
+ */
+export function $autoRoute(context: DecoratorContext, entity: Type, ...args: unknown[]) {
+  validateDecoratorParamCount(context, 0, 0, args);
```

Comment:
Would use the `createDecoratorDefinition` instead of calling those manually. It gives a little better messages and point to the right locations